drivers/timer.c

```diff
@@ -7,11 +7,11 @@ static acpi_gas_t timer_block;
 static int extended_timer = 0;
 static int supported = 0;
 
-static uint32_t read_timer_value(){    
+static uint32_t read_timer_value(){
     if(timer_block.address_space == ACPI_GAS_IO){
         return laihost_ind(timer_block.base);
     } else if(timer_block.address_space == ACPI_GAS_MMIO){
-        uint32_t *reg = (uint32_t *)(timer_block.base);
+        uint32_t *reg = (uint32_t *)((uint32_t) timer_block.base);
```

Comment:
Why are you casting to uin32_t, here? What if the pointer falls in the 64bit address space? If that is required in 32bit mode, maybe cast to `uintptr_t`